### DISCUSSION TITLE
The case of the disappearing buffer

### DISCUSSION THREAD

**Discussion by SpeedyCharly** [AzuraCast/AzuraCast#7461] [General]
Posted this as a bug [#7433] but it didn't attract the attention I had hoped...

Wondering if anyone else has experienced this;

For a long while now, we've been having live streamers getting disconnected for a few seconds without warning following a tiny bit of dead air (usually slight delay in switching to mike after a song). This typically occurs once or twice over the course of their broadcast. Such delays are not usually long enough for the streaming app to go off air. On such occasions, Autodj automatically kicks in with a song that will stop playing as soon as the stream comes back after the little bit of dead air ends.

Looking to the logs was not giving us any obvious clues as to what was happening, so I decided to do some experimentation to hopefully discover what was happening. I would intentionally create little bits of dead air, and, after about 10 or 12 such bits the Autodj would kick in until I resumed streaming (all without having to reconnect as the encoder would stay alive during such a short delay).

After a few times doing this, I started noticing that the buffer delay was getting shorter after each such little bit of dead air. Starting at 30 seconds, it would erode all the way to zero at which time the Autodj would kick in on the next bit of dead air.

I don't remember exactly on which release it started doing this, but it seems to me it wasn't always so; the delay would remain at 30 secs or whatever time you would set the buffer at.

I've included a bit of LS log showing one such occasion where the streamer was dropped - I personally don't see a clue to a solution, but you guys might...

As always, thank you for your hard work and for such a great app
Speedy Charly

LS Log;
024/09/09 17:44:56 [harbor:4] Request to update metadata for mount / on port 8005
2024/09/09 17:44:56 [lang:3] API auth - Sending POST request to 'http://127.0.0.1:6010/api/internal/1/liquidsoap/auth' with body: { "user": "XXX", "password": "YYYY" }
2024/09/09 17:44:57 [lang:3] API auth - Response (200): true
2024/09/09 17:44:57 [harbor:4] Client logged in.
2024/09/09 17:44:57 [input_streamer:3] New metadata chunk ? -- Honoré Godbout - Y'a trop d'amour autour de moi - Speedy en ondes.
2024/09/09 17:45:01 [lang:3] API feedback - Sending POST request to 'http://127.0.0.1:6010/api/internal/1/liquidsoap/feedback' with body: {"title":"Honoré Godbout - Y'a trop d'amour autour de moi - Speedy en ondes"}
2024/09/09 17:45:01 [server:4] New client unix socket "".
2024/09/09 17:45:01 [server:4] Client unix socket "" disconnected.
2024/09/09 17:45:01 [lang:3] API feedback - Response (200): true
2024/09/09 17:46:01 [server:4] New client unix socket "".
2024/09/09 17:46:01 [server:4] Client unix socket "" disconnected.
2024/09/09 17:47:01 [server:4] New client unix socket "".
2024/09/09 17:47:01 [server:4] Client unix socket "" disconnected.
2024/09/09 17:48:01 [server:4] New client unix socket "".
2024/09/09 17:48:01 [server:4] Client unix socket "" disconnected.
2024/09/09 17:48:38 [input_streamer:4] End of track
# !!
# Delay in opening mike at end of song >> switch to Autodj immediately - no buffer left!
# !!
2024/09/09 17:48:38 [input_streamer:4] Buffer emptied, buffering needed. 
# Of course, it got eroded to zero
2024/09/09 17:48:38 [live_fallback:3] Switch to ladspa with forgetful transition.
2024/09/09 17:48:38 [source:4] Source replay_metadata.159 gets down.
2024/09/09 17:48:38 [source:4] Source replay_metadata.160 gets up with content type: {audio=pcm(stereo)}.
2024/09/09 17:48:38 [replay_metadata.160:3] Content type is {audio=pcm(stereo)}.
2024/09/09 17:48:38 [next_song:4] Remaining 0 requests
2024/09/09 17:48:38 [next_song:3] Prepared "/var/azuracast/stations/radio_des_festivals/media/Country Anglo/Freddy Fender - Sweet Summer Day.mp3" (RID 2687).
2024/09/09 17:48:38 [request.2687:4] Currently on air.
2024/09/09 17:48:38 [lang:3] API nextsong - Sending POST request to 'http://127.0.0.1:6010/api/internal/1/liquidsoap/nextsong' with body: 
2024/09/09 17:48:38 [lang:3] API feedback - Sending POST request to 'http://127.0.0.1:6010/api/internal/1/liquidsoap/feedback' with body: {"song_id":"d867c90e776a3afebe1b8e122fb38f2a","artist":"Dani","title":"Un p'tit boy c'est gentil (1970)","playlist_id":"17","media_id":"19267"}
2024/09/09 17:48:38 [lang:3] API nextsong - Response (200): annotate:title="A.L _ L.A",artist="Éric Morel",duration="173.61",song_id="4b2f87ba02a7360a719db709764945cd",media_id="19024",playlist_id="21":media:Vedette/éric_morel_-_a.l___l.a.mp3
2024/09/09 17:48:38 [request.2688:4] Pushed ["annotate:title=\"A.L _ L.A\",artist=\"Éric Morel\",duration=\"173.61\",song_id=\"4b2f87ba02a7360a719db709764945cd\",media_id=\"19024\",playlist_id=\"21\":media:Vedette/éric_morel_-_a.l___l.a.mp3";...].
2024/09/09 17:48:38 [request.2688:4] Resolving "annotate:title=\"A.L _ L.A\",artist=\"Éric Morel\",duration=\"173.61\",song_id=\"4b2f87ba02a7360a719db709764945cd\",media_id=\"19024\",playlist_id=\"21\":media:Vedette/éric_morel_-_a.l___l.a.mp3" (timeout 20s)...
2024/09/09 17:48:38 [request.2688:4] Pushed ["media:Vedette/éric_morel_-_a.l___l.a.mp3";...].
2024/09/09 17:48:38 [request.2688:4] Resolving "media:Vedette/éric_morel_-_a.l___l.a.mp3" (timeout 20s)...
2024/09/09 17:48:38 [request.2688:4] Pushed ["/var/azuracast/stations/radio_des_festivals/media/Vedette/éric_morel_-_a.l___l.a.mp3";...].
2024/09/09 17:48:38 [decoder.video.metadata:4] Unsupported MIME type for "/var/azuracast/stations/radio_des_festivals/media/Vedette/éric_morel_-_a.l___l.a.mp3": audio/mpeg!
2024/09/09 17:48:38 [decoder.ogg.metadata:4] Unsupported MIME type for "/var/azuracast/stations/radio_des_festivals/media/Vedette/éric_morel_-_a.l___l.a.mp3": audio/mpeg!
2024/09/09 17:48:38 [decoder.image.metadata:4] Unsupported MIME type for "/var/azuracast/stations/radio_des_festivals/media/Vedette/éric_morel_-_a.l___l.a.mp3": audio/mpeg!
2024/09/09 17:48:38 [decoder.id3:4] Unsupported file extension for "/var/azuracast/stations/radio_des_festivals/media/Vedette/éric_morel_-_a.l___l.a.mp3"!
2024/09/09 17:48:38 [decoder.id3:4] Unsupported file extension for "/var/azuracast/stations/radio_des_festivals/media/Vedette/éric_morel_-_a.l___l.a.mp3"!
2024/09/09 17:48:38 [decoder.id3:4] Unsupported file extension for "/var/azuracast/stations/radio_des_festivals/media/Vedette/éric_morel_-_a.l___l.a.mp3"!
2024/09/09 17:48:38 [decoder.flac.metadata:4] Unsupported MIME type for "/var/azuracast/stations/radio_des_festivals/media/Vedette/éric_morel_-_a.l___l.a.mp3": audio/mpeg!
2024/09/09 17:48:38 [metadata.mp4:4] Unsupported MIME type for "/var/azuracast/stations/radio_des_festivals/media/Vedette/éric_morel_-_a.l___l.a.mp3": audio/mpeg!
2024/09/09 17:48:38 [decoder.ffmpeg:4] Unsupported file extension for "/var/azuracast/stations/radio_des_festivals/media/Vedette/éric_morel_-_a.l___l.a.mp3"!
2024/09/09 17:48:38 [metadata.flac:4] Unsupported MIME type for "/var/azuracast/stations/radio_des_festivals/media/Vedette/éric_morel_-_a.l___l.a.mp3": audio/mpeg!
2024/09/09 17:48:38 [decoder.ogg:4] Unsupported MIME type for "/var/azuracast/stations/radio_des_festivals/media/Vedette/éric_morel_-_a.l___l.a.mp3": audio/mpeg!
2024/09/09 17:48:38 [decoder.taglib:4] Unsupported file extension for "/var/azuracast/stations/radio_des_festivals/media/Vedette/éric_morel_-_a.l___l.a.mp3"!
2024/09/09 17:48:38 [decoder.ogg:4] Unsupported MIME type for "/var/azuracast/stations/radio_des_festivals/media/Vedette/éric_morel_-_a.l___l.a.mp3": audio/mpeg!
2024/09/09 17:48:38 [decoder.ffmpeg:4] Unsupported file extension for "/var/azuracast/stations/radio_des_festivals/media/Vedette/éric_morel_-_a.l___l.a.mp3"!
2024/09/09 17:48:38 [decoder.mad:4] Unsupported file extension for "/var/azuracast/stations/radio_des_festivals/media/Vedette/éric_morel_-_a.l___l.a.mp3"!
2024/09/09 17:48:38 [decoder.flac:4] Unsupported MIME type for "/var/azuracast/stations/radio_des_festivals/media/Vedette/éric_morel_-_a.l___l.a.mp3": audio/mpeg!
2024/09/09 17:48:38 [decoder.mp4:4] Unsupported MIME type for "/var/azuracast/stations/radio_des_festivals/media/Vedette/éric_morel_-_a.l___l.a.mp3": audio/mpeg!
2024/09/09 17:48:38 [decoder.aac:4] Unsupported MIME type for "/var/azuracast/stations/radio_des_festivals/media/Vedette/éric_morel_-_a.l___l.a.mp3": audio/mpeg!
2024/09/09 17:48:38 [decoder.midi:4] Unsupported MIME type for "/var/azuracast/stations/radio_des_festivals/media/Vedette/éric_morel_-_a.l___l.a.mp3": audio/mpeg!
2024/09/09 17:48:38 [decoder.srt:4] Unsupported MIME type for "/var/azuracast/stations/radio_des_festivals/media/Vedette/éric_morel_-_a.l___l.a.mp3": audio/mpeg!
2024/09/09 17:48:38 [decoder.aiff:4] Unsupported MIME type for "/var/azuracast/stations/radio_des_festivals/media/Vedette/éric_morel_-_a.l___l.a.mp3": audio/mpeg!
2024/09/09 17:48:38 [decoder.wav:4] Unsupported MIME type for "/var/azuracast/stations/radio_des_festivals/media/Vedette/éric_morel_-_a.l___l.a.mp3": audio/mpeg!
2024/09/09 17:48:38 [decoder:4] Available decoders: ffmpeg (priority: 10), mad (priority: 1), image (priority: 1)
2024/09/09 17:48:38 [decoder:4] Trying decoder "ffmpeg"
2024/09/09 17:48:38 [decoder.ffmpeg:3] Requested content-type for "/var/azuracast/stations/radio_des_festivals/media/Vedette/éric_morel_-_a.l___l.a.mp3": {audio=pcm(stereo)}
2024/09/09 17:48:38 [decoder.ffmpeg:3] FFmpeg recognizes "/var/azuracast/stations/radio_des_festivals/media/Vedette/éric_morel_-_a.l___l.a.mp3" as audio: {codec: mp3, 44100Hz, 2 channel(s)}
2024/09/09 17:48:38 [decoder.ffmpeg:3] Decoded content-type for "/var/azuracast/stations/radio_des_festivals/media/Vedette/éric_morel_-_a.l___l.a.mp3": {audio=pcm(stereo)}
2024/09/09 17:48:38 [decoder:4] Selected decoder ffmpeg for file "/var/azuracast/stations/radio_des_festivals/media/Vedette/éric_morel_-_a.l___l.a.mp3" with expected kind {audio=pcm(stereo)} and detected content {audio=pcm(stereo)}
2024/09/09 17:48:38 [next_song:4] Queued 1 requests
2024/09/09 17:48:38 [lang:3] API feedback - Sending POST request to 'http://127.0.0.1:6010/api/internal/1/liquidsoap/feedback' with body: {"song_id":"facd4f25e9873385dc3cc1564a8713ff","artist":"Freddy Fender","title":"Sweet Summer Day","playlist_id":"14","media_id":"20351"}
2024/09/09 17:48:38 [lang:3] API feedback - Response (200): true
2024/09/09 17:48:39 [lang:3] API feedback - Response (200): true
2024/09/09 17:49:01 [server:4] New client unix socket "".
2024/09/09 17:49:01 [server:4] Client unix socket "" disconnected.
# !!
# Return to live feed after 3 seconds of streamer silence - not even long enough for encoder to disconnect!
# Returns with metadata from previous song ? Shouldn't it display XXX is live?
# Returns to normal operations at end of voice segment with buffer restored to 30 seconds
# !!
2024/09/09 17:49:11 [live_fallback:3] Switch to insert_metadata with transition.
2024/09/09 17:49:11 [source:4] Source replay_metadata.160 gets down.
2024/09/09 17:49:11 [lang:3] executing transition to live
2024/09/09 17:49:11 [lang:3] Inserting last live meta: [("pass", "XXXX:YYYY"), ("metadata_url", "https://radiodesfestivals.ca"), ("title", "Honoré Godbout - Y'a trop d'amour autour de moi - Speedy en ondes"), ("is_live", "true")]
2024/09/09 17:49:11 [source:4] Source replay_metadata.161 gets up with content type: {audio=pcm(stereo)}.
2024/09/09 17:49:11 [replay_metadata.161:3] Content type is {audio=pcm(stereo)}.
2024/09/09 17:49:11 [next_song:4] Finished with "/var/azuracast/stations/radio_des_festivals/media/Country Anglo/Freddy Fender - Sweet Summer Day.mp3".
2024/09/09 17:49:11 [request.2687:4] Request finished.
2024/09/09 17:49:11 [lang:3] API feedback - Sending POST request to 'http://127.0.0.1:6010/api/internal/1/liquidsoap/feedback' with body: {"title":"Honoré Godbout - Y'a trop d'amour autour de moi - Speedy en ondes"}
2024/09/09 17:49:11 [lang:3] API feedback - Response (200): true
2024/09/09 17:50:01 [server:4] New client unix socket "".
2024/09/09 17:50:01 [server:4] Client unix socket "" disconnected.

**Comment by BusterNeece** [comment]
Just a little inside baseball on why this likely didn't attract our attention for fixing or addressing it via AzuraCast is because it smells about 99% likely to be a Liquidsoap issue.

**Reply by SpeedyCharly** [reply]
Ok will bounce it to their side of the court then... Thanks

**Comment by n1ce-0ne** [comment]
i had a similar issue & also feel its liquidsoap related.

with my setup a live dj session would suffer a second or 2 dropout back to autodj then straight back
to the live dj stream, seemed to happen about every 40 - 50 minutes so also the session recordings were ruined,
always seemed like it was autodj that had a gap in audio like it always seemed it was the autodj track had the quiet spot which triggered it.

the ONLY way i found to get around it was to cut liquidsoap out & stream directly to the icecast server which is perfect.
